hcloud/instances_test.go

```diff
@@ -644,6 +645,25 @@ func TestNodeAddressesRobotServer(t *testing.T) {
 				{Type: corev1.NodeExternalIP, Address: "203.0.113.7"},
 			},
 		},
+		{
+			name:          "public ipv4 and internal ip",
```

Comment:
Could you add test cases for the conditions where we do not add the IPs? ie `MisconfiguredInternalIP`. Its enough for me to validate that they are not in the returned IPs, no need to check that the event is emitted.